### PR TITLE
Add media folder to perimeter

### PR DESCRIPTION
**Pull request by itkg-nanne**
https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/2094
https://github.com/open-orchestra/open-orchestra-media-bundle/pull/247
https://github.com/open-orchestra/open-orchestra-media-admin-bundle/pull/343

### CODE REVIEW

File: Backoffice/GeneratePerimeter/Strategy/GeneratePerimeterStrategy.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+abstract class GeneratePerimeterStrategy
+{
+    protected $contextManager;
+
+    /**
+     * @param CurrentSiteIdInterface $contextManager
+     */
+    public function __construct(
+        CurrentSiteIdInterface $contextManager
+
```
Comment (ngilain) [inline Backoffice/GeneratePerimeter/Strategy/GeneratePerimeterStrategy.php]: empty line to remove

File: Backoffice/GeneratePerimeter/Strategy/NodeGeneratePerimeterStrategy.php [hunk 2]
```diff
@@ -44,10 +44,9 @@ public function getType()
      */
     public function generatePerimeter()
     {
-        $treeNodes = $this->nodeRepository->findTreeNode($this->contextManager->getCurrentSiteId(), $this->contextManager->getUserCurrentSiteDefaultLanguage(), NodeInterface::ROOT_NODE_ID);
-        $treeNodes = $this->formatPerimeter($treeNodes[0]);
+        $treeNodes = $this->nodeRepository->findTreeNode($this->contextManager->getCurrentSiteId(), $this->contextManager->getUserCurrentSiteDefaultLanguage(), NodeInterface::ROOT_PARENT_ID);
 
-        return $treeNodes;
+        return $this->generateTreePerimeter(is_array($treeNodes) ? $treeNodes : array());
```
Comment (ngilain) [inline Backoffice/GeneratePerimeter/Strategy/NodeGeneratePerimeterStrategy.php]: nodeRepository->findTreeNode always return an array, event an empty one, so it seems this test is useless

File: Backoffice/GeneratePerimeter/Strategy/NodeGeneratePerimeterStrategy.php [hunk 3]
```diff
@@ -57,11 +56,9 @@ public function generatePerimeter()
      */
     public function getPerimeterConfiguration()
     {
-        $treeNodes = $this->nodeRepository->findTreeNode($this->contextManager->getCurrentSiteId(), $this->contextManager->getUserCurrentSiteDefaultLanguage(), NodeInterface::ROOT_NODE_ID);
-
-        $treeNodes = (is_array($treeNodes)) ? $this->formatConfiguration($treeNodes[0]) : array();
+        $treeNodes = $this->nodeRepository->findTreeNode($this->contextManager->getCurrentSiteId(), $this->contextManager->getUserCurrentSiteDefaultLanguage(), NodeInterface::ROOT_PARENT_ID);
 
-        return $treeNodes;
+        return $this->getTreePerimeterConfiguration(is_array($treeNodes) ? $treeNodes : array());
```
Comment (ngilain) [inline Backoffice/GeneratePerimeter/Strategy/NodeGeneratePerimeterStrategy.php]: same here about the return value of nodeRepository->findTreeNode

**Comment by itkg-nanne** [comment]
uppdate done @ngilain

**Comment by ngilain** [comment]
:+1:

**Pull request merged**